Amqp.V1/Akka.Streams.Amqp.V1/AmqpSourceStage.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Akka.Streams.Amqp.V1.Util;
+using Akka.Streams.Stage;
+using Akka.Streams.Supervision;
+using Amqp;
+using Amqp.Framing;
+using Amqp.Types;
+using System;
+
+namespace Akka.Streams.Amqp.V1
+{
+    public sealed class AmqpSourceStage<T> : GraphStage<SourceShape<T>>
+    {
+        public AmqpSourceStage(IAmqpSourceSettings<T> amqpSourceSettings)
+        {
+            AmqpSourceSettings = amqpSourceSettings;
+        }
+
+        public override SourceShape<T> Shape => new SourceShape<T>(Out);
+
+        public Outlet<T> Out { get; } = new Outlet<T>("AmqpSource.Out");
+        public IAmqpSourceSettings<T> AmqpSourceSettings { get; }
+
+        protected override GraphStageLogic CreateLogic(Attributes inheritedAttributes) => new AmqpSourceStageLogic(this, inheritedAttributes);
+
+        private class AmqpSourceStageLogic : GraphStageLogic
+        {
+            private readonly Outlet<T> outlet;
+            private readonly IAmqpSourceSettings<T> ampqSourceSettings;
+            private readonly ReceiverLink receiver;
+            private readonly Decider decider;
+
+            public AmqpSourceStageLogic(AmqpSourceStage<T> stage, Attributes attributes) : base(stage.Shape)
+            {
+                outlet = stage.Out;
+                ampqSourceSettings = stage.AmqpSourceSettings;
+                receiver = stage.AmqpSourceSettings.GetReceiverLink();
+                decider = attributes.GetDeciderOrDefault();
+
+                SetHandler(outlet, () => {
+                    var message = receiver.Receive();
+                    if (message != null)
+                    {
+                        try
+                        {
+                            Push(outlet, ampqSourceSettings.Convert(message));
+                            receiver.Accept(message);
+                        }
+                        catch (Exception e)
+                        {
+                            receiver.Reject(message, new Error(new Symbol(e.Message)));
```

Comment:
Not sure about the semantics of `Reject` but would it maybe make sense to sync it with the result of the decider ?  
E.g. if decider returns `Resume` or `Restart` one maybe doesn't want to reject the message but instead try it another time ?